src/client/src/system/openFin/openFin.js

```diff
@@ -22,16 +22,15 @@ export default class OpenFin {
     return typeof fin !== 'undefined';
   }
 
-  close(){
-    this._currentWindow.close();
+  close(window = this._currentWindow){
+    window.close(true, () => _log.info('Window closed with success.'), err => _log.error(`Failed to close window. Error: ${err}`));
   }
-  
-  minimise(e){
-    this._currentWindow.minimize();
+
+  minimize(window = this._currentWindow){
+   window.minimize(() => _log.info('Window minimized with success.'), err => _log.error(`Failed to minimize window. Error: ${err}`));
```

Comment:
you can dump `err` as the last param and it'll get passed to the debug console, i.e. 
```
log.error(`blah ${err}`, err);
```